File: app/src/main/java/com/stupidtree/hichat/ui/chat/FriendStateTrigger.java
package com.stupidtree.hichat.ui.chat;

import com.stupidtree.hichat.ui.base.Trigger;

public class FriendStateTrigger extends Trigger {
    String id;
    Boolean online;

    public static FriendStateTrigger getActioning(String id,boolean online){
        FriendStateTrigger ft = new FriendStateTrigger();
        ft.online = online;
        ft.id = id;
        ft.setActioning();
        return ft;
    }

    public static FriendStateTrigger getStill(){
        FriendStateTrigger ft = new FriendStateTrigger();
        ft.cancelActioning();
        return ft;
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/chat/FriendStateTrigger.kt
package com.stupidtree.cloudliter.ui.chat

import com.stupidtree.component.data.Trigger

class FriendStateTrigger : Trigger() {
    var conversationId: String? = null
    var online: String? = null
    var num:Int = 0

    companion object {
        fun getActioning(id: String?, online: String?,num:Int): FriendStateTrigger {
            val ft = FriendStateTrigger()
            ft.online = online
            ft.conversationId = id
            ft.num = num
            ft.setActioning()
            return ft
        }

        val still: FriendStateTrigger
            get() {
                val ft = FriendStateTrigger()
                ft.cancelActioning()
                return ft
            }
    }
}